pkg/didcomm/protocol/exchange/exchange.go

```diff
@@ -68,6 +68,11 @@ type Connection struct {
 	DIDDoc *did.Doc `json:"did_doc,omitempty"`
 }
 
+// Provider for exchange protocol
+type Provider interface {
```

Comment:
This should not be imported outside this package. It is the dependencies of this package.
Context should contain the needed methods that match up with the interface naming.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/f4a9ffc938e46fb640514954eef882c1518e1871..463122c87b11e27a303f59b42906628162e71b73#diff-4d3ca323f9b237cfa9940c12efa9b615R72)